src/components/NewCommentForm.tsx

```diff
@@ -1,8 +1,73 @@
-import React from 'react';
+import React, { FormEvent, useState } from 'react';
+import cn from 'classnames';
+import { useComment } from '../hooks/useComment';
+import { usePost } from '../hooks/usePost';
 
 export const NewCommentForm: React.FC = () => {
+  const { addComment } = useComment();
+  const { selectedPost } = usePost();
+
+  const [inputName, setInputName] = useState('');
+  const [inputEmail, setInputEmail] = useState('');
+  const [inputText, setInputText] = useState('');
+
+  const [nameError, setNameError] = useState(false);
+  const [emailError, setEmailError] = useState(false);
+  const [textError, setTextError] = useState(false);
+
+  const [loading, setLoading] = useState(false);
+
+  const validate = () => {
+    if (!inputName.trim()
+      || !inputText.trim()
+      || !inputEmail.trim()
+    ) {
+      throw new Error('please fill the form correctly');
+    }
+
+    return true;
+  };
+
+  const handleSubmit = async (event: FormEvent) => {
+    event.preventDefault();
+    try {
+      setLoading(true);
+
+      if (validate() && selectedPost) {
+        await addComment({
+          name: inputName,
+          email: inputEmail,
+          postId: selectedPost?.id,
+          body: inputText,
+        });
+      }
+
+      setInputText('');
+    } catch (error) {
+      setNameError(!inputName.trim());
+      setEmailError(!inputEmail.trim());
+      setTextError(!inputText.trim());
+    } finally {
+      setLoading(false);
+    }
+  };
```

Comment:
Maybe it's better to setLoading before the try block, and setInputText in the finally block